File: frontend/src/stores/weixin.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { fetchJSON, asArray } from '@/api/client'

export interface WeixinAccount {
  accountId: string
  baseUrl?: string
  userId?: string
  createdAt?: string
  updatedAt?: string
}

export interface WeixinStatus {
  enabled: boolean
  running: boolean
  defaultProjectId?: string
  defaultAgentId?: string
  defaultModelId?: string
  autoApprove: boolean
  accounts: WeixinAccount[]
  bindingCount: number
}

export interface WeixinBinding {
  id: string
  accountId: string
  peerUserId: string
  sessionId: string
  contextToken?: string
}

export interface WeixinLoginStart {
  sessionKey: string
  qrcodeUrl: string
}

export interface WeixinLoginWait {
  connected: boolean
  alreadyConnected?: boolean
  accountId?: string
  userId?: string
  message?: string
  needsVerifyCode?: boolean
}

export const useWeixinStore = defineStore('weixin', () => {
  const status = ref<WeixinStatus | null>(null)
  const bindings = ref<WeixinBinding[]>([])
  const loading = ref(false)
  const saving = ref(false)
  const qr = ref<{ sessionKey: string; url: string } | null>(null)
  const loginMessage = ref('')
  const verifyCode = ref('')

  async function refreshStatus() {
    loading.value = true
    try {
      status.value = await fetchJSON<WeixinStatus>('/channels/weixin/status')
    } finally {
      loading.value = false
    }
  }

  async function refreshBindings() {
    try {
      bindings.value = asArray(await fetchJSON<WeixinBinding[]>('/channels/weixin/bindings'))
    } catch {
      bindings.value = []
    }
  }

  async function configure(payload: {
    enabled: boolean
    defaultAgentId: string
    defaultModelId?: string
    autoApprove?: boolean
  }) {
    saving.value = true
    try {
      status.value = await fetchJSON<WeixinStatus>('/channels/weixin', {
        method: 'PUT',
        body: JSON.stringify(payload),
      })
      await refreshBindings()
      return status.value
    } finally {
      saving.value = false
    }
  }

  async function startLogin() {
    loginMessage.value = ''
    const res = await fetchJSON<WeixinLoginStart>('/channels/weixin/login/start', {
      method: 'POST',
      body: '{}',
    })
    qr.value = { sessionKey: res.sessionKey, url: res.qrcodeUrl }
    return res
  }

  async function waitLogin(timeoutMs = 120000) {
    if (!qr.value) throw new Error('no login session')
    const res = await fetchJSON<WeixinLoginWait>('/channels/weixin/login/wait', {
      method: 'POST',
      body: JSON.stringify({
        sessionKey: qr.value.sessionKey,
        verifyCode: verifyCode.value || undefined,
        timeoutMs,
      }),
    })
    loginMessage.value = res.message || ''
    if (res.connected) {
      qr.value = null
      verifyCode.value = ''
      await refreshStatus()
      await refreshBindings()
    }
    return res
  }

  async function logout(accountId?: string) {
    await fetchJSON('/channels/weixin/logout', {
      method: 'POST',
      body: JSON.stringify({ accountId: accountId || '' }),
    })
    await refreshStatus()
    await refreshBindings()
  }

  function isWeixinSession(sessionId: string): boolean {
    return bindings.value.some((b) => b.sessionId === sessionId)
  }

  return {
    status,
    bindings,
    loading,
    saving,
    qr,
    loginMessage,
    verifyCode,
    refreshStatus,
    refreshBindings,
    configure,
    startLogin,
    waitLogin,
    logout,
    isWeixinSession,
  }
})

File: core/domain/weixin.go
package domain

import "time"

// ConfigChannelsSection holds external chat channel settings.
type ConfigChannelsSection struct {
	Weixin ConfigWeixinChannel `json:"weixin" mapstructure:"weixin" yaml:"weixin"`
}

// ConfigWeixinChannel configures the Weixin iLink bridge.
// Project binding lives on each WeixinAccount (one account → one project).
type ConfigWeixinChannel struct {
	Enabled        bool   `json:"enabled" mapstructure:"enabled" yaml:"enabled"`
	DefaultAgentID string `json:"defaultAgentId" mapstructure:"default_agent_id" yaml:"default_agent_id"`
	DefaultModelID string `json:"defaultModelId" mapstructure:"default_model_id" yaml:"default_model_id"`
	AutoApprove    bool   `json:"autoApprove" mapstructure:"auto_approve" yaml:"auto_approve"`

	// DefaultProjectID is deprecated (migrated onto WeixinAccount.ProjectID).
	// Kept only so one-shot migration can read old YAML.
	DefaultProjectID string `json:"defaultProjectId,omitempty" mapstructure:"default_project_id" yaml:"default_project_id,omitempty"`
}

// WeixinAccount is a logged-in iLink bot account bound to one Teams project.
type WeixinAccount struct {
	AccountID string    `json:"accountId"`
	Token     string    `json:"token,omitempty"`
	BaseURL   string    `json:"baseUrl,omitempty"`
	UserID    string    `json:"userId,omitempty"`
	ProjectID string    `json:"projectId,omitempty"`
	SyncBuf   string    `json:"-"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

// WeixinBinding maps one Weixin peer to one Teams session (1:1).
type WeixinBinding struct {
	ID           string    `json:"id"`
	AccountID    string    `json:"accountId"`
	PeerUserID   string    `json:"peerUserId"`
	SessionID    string    `json:"sessionId"`
	ContextToken string    `json:"contextToken,omitempty"`
	CreatedAt    time.Time `json:"createdAt"`
	UpdatedAt    time.Time `json:"updatedAt"`
}

type WeixinLoginStartResult struct {
	SessionKey string `json:"sessionKey"`
	QRCodeURL  string `json:"qrcodeUrl"`
	AccountID  string `json:"accountId,omitempty"`
}

type WeixinLoginWaitResult struct {
	Connected        bool   `json:"connected"`
	AlreadyConnected bool   `json:"alreadyConnected,omitempty"`
	AccountID        string `json:"accountId,omitempty"`
	UserID           string `json:"userId,omitempty"`
	ProjectID        string `json:"projectId,omitempty"`
	Message          string `json:"message,omitempty"`
	NeedsVerifyCode  bool   `json:"needsVerifyCode,omitempty"`
}

type WeixinStatus struct {
	Enabled        bool            `json:"enabled"`
	Running        bool            `json:"running"`
	DefaultAgentID string          `json:"defaultAgentId,omitempty"`
	DefaultModelID string          `json:"defaultModelId,omitempty"`
	AutoApprove    bool            `json:"autoApprove"`
	Accounts       []WeixinAccount `json:"accounts"`
	BindingCount   int             `json:"bindingCount"`
}
